docs/content/tutorial/step_07.ngdoc

```diff
@@ -1,380 +1,311 @@
+constructor function, if you were to [minify][minification] the JavaScript code for the
+`PhoneListController` controller, all of its function arguments would be minified as well, and the
+dependency injector would not be able to identify services correctly.
 
-<div style="display:none">
-TODO!
-<img  class="diagram" src="img/tutorial/tutorial_07_final.png">
-</div>
+We can overcome this problem by annotating the function with the names of the dependencies, provided
+as strings, which will not get minified. There are two ways to provide these injection annotations:
 
-We also added a placeholder template for the phone details view:
+* Create an `$inject` property on the controller function which holds an array of strings.
+  Each string in the array is the name of the service to inject for the corresponding parameter.
+  In our example, we would write:
 
-__`app/partials/phone-detail.html`:__
+  ```js
+      function PhoneListController($http) {...}
+      PhoneListController.$inject = ['$http'];
+      ...
+      .component('phoneList', {..., controller: PhoneListController});
+  ```
 
-```html
-TBD: detail view for <span>{{phoneId}}</span>
-```
+* Use an inline annotation where, instead of just providing the function, you provide an array.
+  This array contains a list of the service names, followed by the function itself as the last item
+  of the array.
 
-Note how we are using the `phoneId` expression which will be defined in the `PhoneDetailCtrl` controller.
+  ```js
+      function PhoneListController($http) {...}
+      ...
+      .component('phoneList', {..., controller: ['$http', PhoneListController]});
+  ```
 
-## The App Module
+Both of these methods work with any function that can be injected by Angular, so it's up to your
+project's style guide to decide which one you use.
 
-To improve the organization of the app, we are making use of Angular's `ngRoute` module and we've
-moved the controllers into their own module `phonecatControllers` (as shown below).
-
-We added `angular-route.js` to `index.html` and created a new `phonecatControllers` module in
-`controllers.js`. That's not all we need to do to be able to use their code, however. We also have
-to add the modules as dependencies of our app. By listing these two modules as dependencies of
-`phonecatApp`, we can use the directives and services they provide.
-
-
-__`app/js/app.js`:__
+When using the second method, it is common to provide the constructor function inline, when
+registering the controller:
 
 ```js
-var phonecatApp = angular.module('phonecatApp', [
-  'ngRoute',
-  'phonecatControllers'
-]);
-
-...
+.component('phoneList', {..., controller: ['$http', function PhoneListController($http) {...}]});
 ```
 
-Notice the second argument passed to `angular.module`, `['ngRoute', 'phonecatControllers']`. This
-array lists the modules that `phonecatApp` depends on.
+From this point onwards, we are going to use the inline method in the tutorial. With that in mind,
```

Comment:
I don't think annotation style makes much difference. Inline definition does indeed make it more difficult to put stuff on the prototype, but refactoring them is trivial as well: E.g.:

```js
.component('myComp', {
  templateUrl: '...',
  bindings: {...},
  controller: function MyCompController(...) {
    this.someProp = ...;
    this.someMethod = ...;
    ...
  }
})

// to

.component('myComp', {
  templateUrl: '...',
  bindings: {...},
  controller: MyCompController
})

function MyCompController(...) {
  this.someProp = ...;
  ...
}
MyCompController.prototype.someMethod = ...;
...
```

--
Personally, I don't see much benefit in using the prototype for controllers and services. And by the time people get to bother with the difference of defining methods in the constructor vs on the prototype, classes will be all over the place :smiley:

I like how the inline definition looks much more similar to a class (with TypeScript annotation in ng2). I.e.:

```js
.component('myComp', {
  templateUrl: '...',
  bindings: {...},
  controller: function MyCompController(...) {
    this.someProp = ...;
    this.someMethod = ...;
    ...
  }
})

// looks at lot like:

@Component(
  selector: 'myComp',
  templateUrl: '...',
  inputs: [...]
}
class MyCompController {
  this.someProp = ...;
  someMethod() {...}
  ...
}
```

--
Most importantly, for a tutorial, I believe simpler is better.
(Plus, I could definitely do without having to exlain JS prototypal inheritance :grin:)